typescript/sdk/src/token/Token.ts

```diff
@@ -232,24 +214,46 @@ export class Token implements IToken {
         token: addressOrDenom,
       });
     } else if (standard === TokenStandard.SealevelHypNative) {
+      assert(mailbox, `Mailbox required for Sealevel hyp tokens`);
       return new SealevelHypNativeAdapter(
         chainName,
         multiProvider,
-        sealevelAddresses!,
+        {
+          warpRouter: addressOrDenom,
+          mailbox,
+        },
         false,
       );
     } else if (standard === TokenStandard.SealevelHypCollateral) {
+      assert(mailbox, `Mailbox required for Sealevel hyp tokens`);
+      assert(
+        collateralAddressOrDenom,
+        `collateralAddressOrDenom required for Sealevel hyp collateral tokens`,
+      );
       return new SealevelHypCollateralAdapter(
         chainName,
         multiProvider,
-        sealevelAddresses!,
+        {
+          warpRouter: addressOrDenom,
+          token: collateralAddressOrDenom,
+          mailbox,
+        },
         false,
       );
     } else if (standard === TokenStandard.SealevelHypSynthetic) {
+      assert(mailbox, `Mailbox required for Sealevel hyp tokens`);
+      assert(
+        collateralAddressOrDenom,
+        `collateralAddressOrDenom required for Sealevel hyp collateral tokens`,
+      );
```

Comment:
I think this should be omitted here (synthetic branch)

Reply:
I think it's at least needed for balance checks: https://github.com/hyperlane-xyz/hyperlane-monorepo/blob/5a0d68bdce143d31bd775e697aca11119c60fdea/typescript/sdk/src/token/adapters/SealevelTokenAdapter.ts#L157

Reply:
I agree with Dan, I think it's needed b.c. Solana